app/workers/education_form/create_daily_spool_files.rb

```diff
@@ -99,8 +99,15 @@ def write_files(writer, structured_data:)
             writer.write(contents, filename)
 
             # send copy of staging spool files to testers
+            # This mailer is intended to only work for the lower environments and NOT production
+            # Rails.env will return 'production' on the development & staging servers and  which
+            # will trip the unwary. To be safe, use ENV['HOSTNAME']
             email_staging_spool_files(contents) if
+              # local developer development
               Rails.env.eql?('development') ||
+              # local developer test and production is mocked to confirm it does not work
+              (Rails.env.eql?('test') && !ENV['HOSTNAME'].eql?('production-api.va.gov')) ||
```

Comment:
@GcioGregg Just curious - what is the purpose of this line? Is it specifically so that a test will pass? Generally speaking, I try not to make additions tot eh code just for a test's sake, unless it's necessary.

I like your tests below! My understanding of them is that when `Rails.env == 'test'`, deliveries are sent (>0), and when `ENV['HOSTNAME'] === 'production-api.va.gov'`, none are sent. That seems perfect.

So, can we remove the line about `!ENV['HOSTNAME'].eql?('production-api.va.gov')`? It seems to me that if `HOSTNAME` is `'staging-api.va.gov'`, then we know it's not `'production-api.va.gov'`. Maybe I'm missing something?

Reply:
The issue is that we're trying to prove this will NOT run in production. You mentioned something about testing the contra condition (not running in an environment we don't want it to run in). There were some concerns raised by our users about that also.

So we mocked a scenario to try to replicate this running in production. See comment lines 103 & 104 in create_daily_spool_files.rb to understand part of the issue with mocking it. Also see the test starting on line 142 in create_daily_spool_files.spec.rb.

There are two separate settings. Rails.env will either be 'development' or 'test' on your sandbox and 'production' on the development, staging and production servers (to confirm this run a rails c inside a pod on any of the VA Servers and observe what Rails.env returns as a value. Spoiler: it's 'production' but you should see this for yourself). So, to mock running in production you need to look at ENV['HOSTNAME'] which is done in the test referenced above where the Rails.env will return 'test'. We want the method to always execute in the test env EXCEPT  for when we are explicitly mocking production.

Let's go over this on a huddle if you have any further questions about this. Btw, I suspect this business about the Rails.env is tripping up lots of developer teams and awareness about this needs to be raised. Gregg and I saw all over the place code like Rails.env.production? which will return true on servers I don't think people want it to be returning true on. (I.e VA development & VA Staging) There probably needs to be a application wide helper method created to return the correct environment variable in vets-api. There is one for GIDS.